generator/api/src/test/java/io/fabric8/maven/generator/api/FromSelectorTest.java

```diff
@@ -97,26 +97,92 @@ public void simpleJava11() {
+                openshift, docker, "1.2.3.fuse-00009", "redhat-docker-prop", "redhat-istag-prop",
+                openshift, s2i, "1.2.3.foo-00009", "s2i-prop", "istag-prop",
+                openshift, docker, "1.2.3.foo-00009", "docker-prop-8", "istag-prop",
+                openshift, s2i, "1.2.3", "s2i-prop", "istag-prop",
+                openshift, docker, "1.2.3", "docker-prop-8", "istag-prop",
+                null, s2i, "1.2.3.redhat-00009", "redhat-docker-prop", "redhat-istag-prop",
+                null, docker, "1.2.3.redhat-00009", "redhat-docker-prop", "redhat-istag-prop",
+                null, s2i, "1.2.3.fuse-00009", "redhat-docker-prop", "redhat-istag-prop",
+                null, docker, "1.2.3.fuse-00009", "redhat-docker-prop", "redhat-istag-prop",
+                null, s2i, "1.2.3.foo-00009", "docker-prop-8", "istag-prop",
+                null, docker, "1.2.3.foo-00009", "docker-prop-8", "istag-prop",
+                null, s2i, "1.2.3", "docker-prop-8", "istag-prop",
+                null, docker, "1.2.3", "docker-prop-8", "istag-prop",
+                openshift, null, "1.2.3.redhat-00009", "redhat-docker-prop", "redhat-istag-prop",
+                openshift, null, "1.2.3.fuse-00009", "redhat-docker-prop", "redhat-istag-prop",
+                openshift, null, "1.2.3.foo-00009", "docker-prop-8", "istag-prop",
+                openshift, null, "1.2.3", "docker-prop-8", "istag-prop"
+        };
+
+        for (int i = 0; i < data.length; i += 5) {
+            prepareExpectionJava8Target((String) data[i + 2], (RuntimeMode) data[i], (OpenShiftBuildStrategy) data[i + 1]);
+            FromSelector selector = new FromSelector.Default(ctx, "test");
+
+            assertEquals(data[i + 3], selector.getFrom());
+            Map<String, String> fromExt = selector.getImageStreamTagFromExt();
+            assertEquals(fromExt.size(),3);
+            assertEquals(fromExt.get(SourceStrategy.kind.key()), "ImageStreamTag");
+            assertEquals(fromExt.get(SourceStrategy.namespace.key()), "openshift");
+            assertEquals(fromExt.get(SourceStrategy.name.key()), data[i + 4]);
+        }
+    }
+
+    @Test
+    public void Java11BaseImageTarget() {
+        final Object[] data = new Object[] {
+                openshift, docker, "1.2.3.foo-00009", "docker-prop-11",
+                openshift, docker, "1.2.3", "docker-prop-11",
+                null, s2i, "1.2.3.foo-00009", "docker-prop-11",
+                null, docker, "1.2.3.foo-00009", "docker-prop-11",
+                openshift, null, "1.2.3.foo-00009", "docker-prop-11",
+                openshift, null, "1.2.3", "docker-prop-11",
+        };
+
+        for (int i = 0; i < data.length; i += 4) {
+            prepareExpectionJava11Target((String) data[i + 2], (RuntimeMode) data[i], (OpenShiftBuildStrategy) data[i + 1]);
+            FromSelector selector = new FromSelector.Default(ctx, "test");
+
+            assertEquals(data[i + 3], selector.getFrom());
```

Comment:
Why are we only asserting the value of `selector.getFrom()` in Java11 tests? I see some assertions related to `ImageStreamTag` in Java8 tests. Is it possible to add those assertions here to make this test more concrete?

Reply:
ImageStreamTag is for Openshift mode. Java 11 isn't available for that. also, this has been the case before this PR as well.

Reply:
Sure about that? I can see Java11 tags for fabric8-s2i image: https://hub.docker.com/layers/fabric8/s2i-java/latest-java11/images/sha256-a7add25f788067be62d8970eec771d7a237ecc6905c64ee8bb898ad45c90c540

Reply:
Yes, but fmp doesn't support java 11 images for s2i yet. I can add that in a separate PR.